test/testinfra.py

```diff
@@ -170,66 +208,44 @@ def prioritize(self, update=False):
                 whitelist = [ user["login"] ]
 
-        master = self.get("git/refs/heads/master")
-        context = self.context()
+        results = []
+
+        if update:
+            status = { "state": "pending", "description": "Not yet tested", "context": self.context() }
+        else:
+            status = None
 
-        # Add master with a priority of 11
-        results = [(9, "master", master["object"]["sha"], "master")]
+        master = self.get("git/refs/heads/master")
+        priority = self.prioritize(master["object"]["sha"], update=status, baseline=9)
+        if priority > 0:
+            results.append((priority, "master", master["object"]["sha"], "master"))
 
         # Load all the pull requests
         for pull in self.get("pulls"):
-            state = None
-            last = { }
-            statuses = self.get("commits/{0}/statuses".format(pull["head"]["sha"]))
-            if statuses:
-                for status in statuses:
-                    if status["context"] == context:
-                        state = status["state"]
-                        last = status
-                        break
-
-            if state in [ "success", "failure" ]:
-                continue
-
-            # Pull in the labels for this pull
-            labels = []
-            for label in self.get("issues/{0}/labels".format(pull["number"])):
-                labels.append(label["name"])
-
-            status = { "state": "pending", "description": "Not yet tested", "context": context }
-            priority = 10
-
-            if "priority" in labels:
-                priority += 2
-            if "needsdesign" in labels:
-                priority -= 2
-            if "needswork" in labels:
-                priority -= 3
+            baseline = 10
 
             # It needs to be in the whitelist
             login = pull["head"]["user"]["login"]
             if login not in whitelist:
                 status["description"] = "Manual testing required"
-                priority = 0
+                baseline = 0
 
-            # Is testing already in progress?
-            if last.get("description", None) == TESTING:
-                priority = 0
+            # Pull in the labels for this pull
+            labels = []
+            for label in self.get("issues/{0}/labels".format(pull["number"])):
+                labels.append(label["name"])
 
-            revision = pull["head"]["sha"]
             number = pull["number"]
-            ref = "pull/%d/head" % number
-
-            if update and not dict_is_subset(last, status):
-                self.post("statuses/" + revision, status)
+            revision = pull["head"]["sha"]
 
+            priority = self.prioritize(revision, labels, update=status, baseline=baseline)
             if priority > 0:
-                results.append((priority, "pull-" + str(number), revision, ref))
+                results.append((priority, "pull-%d" % number, revision, "pull/%d/head" % number))
```

Comment:
why mix different string formatting styles in the same pull request? should we care which one is used?